cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -2453,3 +2484,46 @@ def revert_to_snapshot(self, context, volume, snapshot):
             LOG.debug("Backing does not exist for volume.", resource=volume)
         else:
             self.volumeops.revert_to_snapshot(backing, snapshot.name)
+
+    def migrate_volume(self, context, volume, host):
+        """Migrate a volume to the specified host.
+
+        If the backing is not created, the dest host will create it.
+        """
+        backing = self.volumeops.get_backing(volume.name, volume.id)
+        dest_host = host['host']
+        # If the backing is not yet created, let the destination host create it
+        # In this case, migration is not necessary
+        if not backing:
+            LOG.info("There is no backing for the volume: %(volume_name)s. "
+                     "Creating it on dest_host %(dest_host)s.",
+                     {'volume_name': volume.name, 'dest_host': dest_host})
+            self._remote_api.create_backing(context, host['host'], volume)
+            return (True, None)
+
+        service_locator = self._remote_api.get_service_locator_info(context,
+                                                                    dest_host)
```

Comment:
Other drivers parse stuff from host['capabilities']['location_info']. Can we use that instead? What's missing?

Reply:
My only concern is that, we would send the vCenter credentials (username and password) in the capabilities. Not sure if capabilities might be exposed in other places, like logs?

Reply:
I haven't looked through all the drivers, yet. Some send at least a user with the location_info. Having different passwords per vCenter seems like the big problem here, yes. Just wanted to confirm my suspicion, thanks.